rclpy/test/test_executor.py

```diff
@@ -68,6 +69,15 @@ def test_add_node_to_executor(self):
         executor.add_node(self.node)
         self.assertIn(self.node, executor.get_nodes())
 
+    def test_executor_spin_non_blocking(self):
+        self.assertIsNotNone(self.node.handle)
+        executor = SingleThreadedExecutor()
+        executor.add_node(self.node)
+        start = time.time()
```

Comment:
for consistency I'd use steady time (`time.monotonic`) here given that this is what the rcl timers are using

Reply:
Done in ac6250fcd304f3e7d5f283126bd7ac3322543238